Rakefile

```diff
@@ -196,22 +197,22 @@ namespace :toolchain do
   desc 'Sync Rust toolchain to all sources'
   task sync: %i[sync:manifests sync:ci]
 
-  rust_toolchain = File.read('rust-toolchain').chomp
+  rust_toolchain = TOML.load_file('rust-toolchain.toml')
+  toolchain_version = rust_toolchain['toolchain']['channel']
 
   namespace :sync do
     desc 'Sync the root rust-toolchain version to all crate manifests'
     task :manifests do
-      rust_version = File.read('rust-toolchain').chomp
       regexp = /^rust-version = "(.*)"$/
-      next_rust_version = "rust-version = \"#{rust_toolchain}\""
+      next_rust_version = "rust-version = \"#{toolchain_version}\""
 
       pkg_files = FileList.new(['Cargo.toml', 'fuzz/Cargo.toml', 'spec-runner/Cargo.toml', 'ui-tests/Cargo.toml'])
 
       failures = pkg_files.map do |file|
         contents = File.read(file)
 
         if (existing_version = contents.match(regexp))
-          File.write(file, contents.gsub(regexp, next_rust_version)) if existing_version != rust_version
+          File.write(file, contents.gsub(regexp, next_rust_version)) if existing_version != next_rust_version
```

Comment:
`existing_version` contains the full regexp string (e.g. `rust-version = "1.70.0"`), so we should  match against the full `next_rust_version`, rather than just the toolchain version.

Reply:
Yup this is intentional, just in case we ever have a crate version that has the same version string as the Rust toolchain version.